velox/exec/HashPartitionFunction.h

```diff
@@ -58,4 +58,31 @@ class HashPartitionFunction : public core::PartitionFunction {
   SelectivityVector rows_;
   raw_vector<uint64_t> hashes_;
 };
+
+class HashPartitionFunctionSpec : public core::PartitionFunctionSpec {
+ public:
+  HashPartitionFunctionSpec(
+      RowTypePtr inputType,
+      std::vector<column_index_t> keys,
+      std::vector<VectorPtr> constValues)
```

Comment:
do not abbreviate and use keys instead of values (these are values of the keys, but using just values is confusing)

constantKeys